### PR TITLE
Improve `cabal init` code a bit

### PR DESCRIPTION
**Pull request by phadej**
ping @m-renaud

### CODE REVIEW

File: cabal-install/Distribution/Client/Init/Command.hs [hunk 1]
```diff
@@ -269,39 +267,45 @@ getVersion flags = do
...
-         ?>> fmap (fmap (either UnknownLicense id))
-                  (maybePrompt flags
-                    (promptList "Please choose a license" listedLicenses
-                     (Just BSD3) displayLicense True))
-
-  case checkLicenseInvalid lic of
-    Just msg -> putStrLn msg >> getLicense flags
-    Nothing  -> return $ flags { license = maybeToFlag lic }
-
+  elic <-     return (fmap Right $ flagToMaybe $ license flags)
```
Comment (m-renaud) [inline cabal-install/Distribution/Client/Init/Command.hs:270]: Nit: de-indent return (...) by one char to match below.

File: cabal-install/Distribution/Client/Init/Defaults.hs [hunk 2]
```diff
@@ -21,11 +21,11 @@ import Distribution.ModuleName
...
   ( fromString )
-import Distribution.Version
-  ( Version, mkVersion )
+import Distribution.CabalSpecVersion
+  ( CabalSpecVersion (..))
 
-defaultCabalVersion :: Version
-defaultCabalVersion = mkVersion [1,10]
+defaultCabalVersion :: CabalSpecVersion
+defaultCabalVersion = CabalSpecV1_10
```
Comment (m-renaud) [inline cabal-install/Distribution/Client/Init/Defaults.hs:28]: Optional: can you change this to `CabalSpecV2_4` to match other places? I missed the default value in the prompt and only defaulted it under non-interactive init when unspecified.

Comment (phadej) [reply cabal-install/Distribution/Client/Init/Defaults.hs:28]: I'll change non-interactive to actually use values from `Defaults`

Comment (m-renaud) [reply cabal-install/Distribution/Client/Init/Defaults.hs]: SG.

**Pull request merged**